### DISCUSSION TITLE
Add the ability to build a semantic model from a dbt project

### DISCUSSION THREAD

**Discussion by GreyskullAnalytics** [TabularEditor/TabularEditor3#1171] [Ideas]
### Is your feature request related to a problem? Please describe

If I build a model on Databricks I can reuse table metadata to populate table and field descriptions, but not all sources have this feature (I'm yet to find a way to do it with Fabric). However, dbt does allow you to provide this data in the models file

### Describe the solution you'd like

With dbt getting increasing adoption, it would be amazing if I could build a semantic model by pointing at a dbt project. This would allow me to select tables and recycle the descriptions from the models file. You can get data source connection string info via the profiles file to authenticate to the actual source to get data types. In time you could even start to recycle dbt semantic layer objects to build DAX measures.

### Goals

1. Awesome portability. dbt models can be built on many different source systems. By abstracting the connection from the actual source to a dbt model it will become far easier to migrate models between different sources (say if someone migrated from Databricks to Fabric)
2. Reuse of dbt documentation - similar to importing descriptions from SQL Server extended properties or using Databricks DESCRIBE, you could flush through the descriptions from the dbt models file to the semantic model description fields
3. It's cool and trendy 😎

### Background

None yet

**Comment by otykier** [comment]
I like this idea, although I'm unsure how many DBT users also use Tabular Editor. I wonder if a short term solution could be using Tabular Editor C# scripts? Are there any .NET SDKs available for working with DBT projects?

**Reply by mlonsk** [reply]
I think we might see more of dbt in Fabric as it is, as far as I know, already integrated through the spark engine. Or at least possible to extent the spark engine with core dbt

**Reply by GreyskullAnalytics** [reply]
dbt models work a bit like serialised source control in TE. Ultimately it's a set of .yaml and .sql files that would be local to the developer
